pkg/vmprovider/providers/vsphere/virtualmachine/backup.go

```diff
@@ -35,51 +35,101 @@ func BackupVirtualMachine(
 	vmCtx context.VirtualMachineContext,
 	vcVM *object.VirtualMachine,
 	bootstrapData map[string]string) error {
```

Comment:
To avoid circular import. The `GetVMMetadata` is currently defined under the `vsphere` package which imports the `virtualmachine` package and calls this backup function.